pkg/csi/service/common/commonco/coagnostic.go

```diff
@@ -35,11 +35,11 @@ type COCommonInterface interface {
 
 // GetContainerOrchestratorInterface returns orchestrator object
 // for a given container orchestrator type
-func GetContainerOrchestratorInterface(ctx context.Context, orchestratorType int, clusterFlavour cnstypes.CnsClusterFlavor) (COCommonInterface, error) {
+func GetContainerOrchestratorInterface(ctx context.Context, orchestratorType int, featureStatesConfigMapInfo config.FeatureStatesConfigInfo) (COCommonInterface, error) {
```

Comment:
`GetContainerOrchestratorInterface` should be the container orchestrator agnostic.

featureStatesConfigMapInfo says it is configmap and thus only applicable to kubernetes. other container orchestrators may not be using config-map to persist app configurations.

Reply:
Done